internal/crdReplicator/crdReplicator-operator.go

```diff
@@ -523,6 +567,16 @@ func (c *Controller) AddedHandler(obj *unstructured.Unstructured, gvr schema.Gro
 		klog.Infof("%s -> resource %s %s of type %s has not a destination label with the ID of the peering cluster", c.ClusterID, obj.GetName(), obj.GetNamespace(), gvr.String())
 		return
 	}
+
+	var remoteNamespace string
+	// if the namespaces is empty, the resource is cluster scoped, so we do no need namespace translations
+	if namespace := obj.GetNamespace(); namespace != "" {
```

Comment:
Could you factorize the code for detecting namespaced/notnamespaced resource? There are several calls in the code...